nearcore/src/runtime/mod.rs

```diff
@@ -122,7 +122,10 @@ impl EpochInfoProvider for SafeEpochManager {
 /// TODO: this possibly should be merged with the runtime cargo or at least reconciled on the interfaces.
 pub struct NightshadeRuntime {
     genesis_config: GenesisConfig,
-    genesis_runtime_config: Arc<RuntimeConfig>,
+    /// Runtime configuration.  Note that it may be slightly different than
+    /// `genesis_config.runtime_config`.  Consider `max_gas_burnt_view` value
+    /// which may be configured per node.
+    runtime_config: Arc<RuntimeConfig>,
```

Comment:
I feel that config handling here accumulated quite a bit of technical debt, as we now have to separate paths that "patch" config: one with `max_gas_burnt`, and another with `RuntimeConfig::from_protocol_version` (which is especcially confusing, as it relies on hidden global state). I have a stalled PR which cleans parts of this up in https://github.com/near/nearcore/pull/4263#issuecomment-832113787.

This isn't strictly related to the PR, so it isn't a blocker, but I want to raise awareness and signal that straightening this up is non-urgent, but important ;-)